scheduler/src/cook/queue_limit.clj

```diff
@@ -0,0 +1,155 @@
+(ns cook.queue-limit
+  (:require [clj-time.core :as time]
+            [clojure.tools.logging :as log]
+            [cook.caches :as caches]
+            [cook.config :as config]
+            [cook.datomic :as datomic]
+            [cook.queries :as queries]
+            [cook.regexp-tools :as regexp-tools]
+            [cook.util :as util]
+            [datomic.api :as d]
+            [plumbing.core :as pc]
+            [metrics.timers :as timers]))
+
+(defn- match-based-on-pool-name
+  "Given a pool name and field name, returns
```

Comment:
Maybe inline this into the caller or rename it? This function having the same name as regexp-tools/match-based-on-pool-name but behaving differently is awkward. 

My vote would be to make match-based-on-pool-name accept a default if not found, and then inline this into its callers, with maybe a helper function 

(def per-pool-config []
  (:per-pool (config/queue-limits)))

Then they become, e.g., :
   (match-based-on-pool-name (per-pool-config) pool-name :pool-global-threshold Integer/MAX_VALUE)

Reply:
Sounds good, will do

Reply:
Done